docs/spec_draft.md

```diff
@@ -1406,6 +1407,69 @@ and produces a `result` tensor. More formally,
 ```
 
 [Back to Ops](#index-of-ops)
+
+## stablehlo.rng_bit_generator
+
+### Semantics
+
+Returns an `output` filled with uniform random data and an updated
+`output_state` using the `rng_algorithm` given `initial_state`. The output is
+guaranteed to be deterministic from the initial state, but it is not guaranteed
+to be deterministic between backends and different compiler versions.
+
+`initial_state` is the initial state of the current random number generation.
+The initial state, required shape, and valid values are dependent on the
+algorithm used.
+
+`rng_algorithm` is one of the following:
+  * `DEFAULT`: Backend specific algorithm.
+  * `THREE_FRY`: Threefry counter-based PRNG algorithm.*
+  * `PHILOX`: Philox algorithm to generate random numbers in parallel.*
+
+\* See: [Salmon et al. SC 2011. Parallel random numbers: as easy as 1, 2, 3.
+](http://www.thesalmons.org/john/random123/papers/random123sc11.pdf)
+
+More formally, given the function `rng_algorithm` which takes an
+`initial_state` as input, the `output` with uniform random data is generated
+given `initial_state`:
+  * `output_state, output = rng(initial_state)`
+
+### Inputs
+
+| Name            | Type                                                 |
+|-----------------|------------------------------------------------------|
+| `initial_state` | tensor of integer or floating-point type             |
+| `rng_algorithm` | constant of type `enum {DEFAULT, THREE_FRY, PHILOX}` |
+
+### Outputs
+
+| Name           | Type                                     |
+|----------------|------------------------------------------|
+| `output_state` | tensor of integer or floating-point type |
+| `output`       | tensor of integer or floating-point type |
+
+### Constraints
+
+  * (C1) Shape of `initial_state` vary depending on the `rng_algorithm`:
+    * If `rng_algorithm = DEFAULT`, `initial_state` has backend-specific
+      requirements.
+    * If `rng_algorithm = THREE_FRY`, `initial_state` is 2-dimensional.
```

Comment:
In the hlo spec we have the element-type also specified u64[2]. But I cannot see the any verification check in MHLO. Can we find the relevance of that element-type?

Reply:
The element type requirement comes from the two algorithms. Indeed, we don't have verification check, so status for verification should be `revisit`. Thanks for pointing this out.

Reply:
Yes, also have a ticket to track it.

Reply:
Added #486